crates/wasmtime/src/runtime/component/get_host.rs

```diff
@@ -0,0 +1,255 @@
+/// `GetHost` generated by [`bindgen!`](super::bindgen).
+///
+/// ```
+/// use wasmtime::component::{Linker, GetHost};
+///
+/// // generated by `bindgen!`
+/// trait Host { /* ... */ }
+/// // generated by `bindgen!`
+/// impl<'a, T: ?Sized + Host> Host for &'a mut T { /* ... */ }
+///
+/// // generated by `bindgen!`
+/// fn add_to_linker_get_host<T, G>(linker: &mut Linker<T>, get_host: G)
+///     where G: for<'a> GetHost<&'a mut T, Host: Host>,
+/// {
+///     // ...
+/// # let _ = (linker, get_host);
+/// }
+///
+/// // In some cases you'll implement `Host` for the `T` in the store directly.
+/// struct MyType;
+///
+/// impl Host for MyType {
+///     // ...
+/// }
+///
+/// fn my_add_to_linker(linker: &mut Linker<MyType>) {
+///     let get_host = type_annotate(|x| x);
+///     add_to_linker_get_host(linker, get_host);
+///
+///     // Type annotating a closure is not the most intuitive thing to do in
+///     // Rust, but that's all this function is doing. This is taking a `F`
+///     // and then simply returning it, but the purpose of this function is to
+///     // synthetically place a bound on `F` here with a `where` clause. This
+///     // `where` clause is what annotates the type of `F` manually and is how
+///     // we accurately describe the lifetimes associated with this closure.
+///     fn type_annotate<F>(f: F) -> F
+///         where F: Fn(&mut MyType) -> &mut MyType
+///     {
+///         f
+///     }
+/// }
+///
+/// // In other cases you might store the structure implementing `Host`
+/// // elsewhere in the store
+/// struct MyContainerType {
+///     the_impl: MyType,
+/// }
+///
+/// fn my_add_to_linker_with_container(linker: &mut Linker<MyContainerType>) {
+///     let get_host = type_annotate(|x| &mut x.the_impl);
+///     add_to_linker_get_host(linker, get_host);
+///
+///     fn type_annotate<F>(f: F) -> F
+///         where F: Fn(&mut MyContainerType) -> &mut MyType
+///     {
+///         f
+///     }
+/// }
+///
+/// // And in some cases you might want to combine pieces of state within a `T`
+/// // of a `Store` into one type that implements this state. This is, for
+/// // example, how WASI impls work in Wasmtime.
+/// struct MyStoreState {
+///     the_table: MyTable,
+///     the_impl: MyType,
+/// }
+///
+/// struct MyTable { /* ... */ }
+///
+/// struct MyBorrowedState<'a> {
+///     the_table: &'a mut MyTable,
+///     the_impl: &'a mut MyType,
+/// }
+///
+/// impl Host for MyBorrowedState<'_> {
+///     // ...
+/// }
+///
+/// fn my_add_to_linker_with_more_lifetimes(linker: &mut Linker<MyStoreState>) {
+///     let get_host = type_annotate(|x| MyBorrowedState {
+///         the_table: &mut x.the_table,
+///         the_impl: &mut x.the_impl,
+///     });
+///     add_to_linker_get_host(linker, get_host);
+///
+///     fn type_annotate<F>(f: F) -> F
+///         where F: Fn(&mut MyStoreState) -> MyBorrowedState<'_>
+///     {
+///         f
+///     }
+/// }
+/// ```
+///
+/// # Too-much Detail
```

Comment:
```suggestion
/// # Too much Detail
```

Nit: remove unnecessary hyphen